docs/checkit.js

```diff
@@ -175,26 +175,40 @@ CheckIt.prototype.displayActiveCalendar = function() {
     
     var checkit = this;
 
-    return checkit.store.getActive().then(function(activeCalendarId) {
-        return checkit.store.loadById(activeCalendarId).catch(function(err) {
-            console.log("There is no current active calendar " + err);
-            checkit.store.removeActive();
-            checkit.showForm(checkit.$buildCalendarForm);
-            return err;
-            });
-        })
-        .then(function(activeCalendarState) {
-            if (activeCalendarState !== null) {
-                var state = activeCalendarState;
-                var calendar = new Calendar(state);
-                return checkit.displayCalendar(calendar);
-            }
+    // if the first promise doesn't work, then just go to catch, don't
+    // try to resolve things inside of a catch and then run .then
+    return checkit.store.getActive()
+        .then(function(activeCalendarId) {
+            // if loadById returns a rejected promise we want to handle it
+            // before running .then, all it does is return a rejected promise
+            // with the reason for rejection, but we can remove the active status
+            // we don't really need .then to run after the catch because everything 
+            // is handled inside of .catch. That's why we are returning a rejected
+            // promise, the .then never gets run. 
+            return checkit.store.loadById(activeCalendarId)
+                // if loadById returns a rejected promise, TODO
+                .catch(function(err) {
+                    console.log("There is no current active calendar " + err);
+                    checkit.store.removeActive();
+                    checkit.showForm(checkit.$buildCalendarForm);
+                    return err;
+                });
         })
+                .then(function(activeCalendarState) {
```

Comment:
It looks like this is attached to `getActive()`, is that what you intended?

`getActive().then(....).then(....).catch(..);` ?